bin/scripts/generate_translations.sh

```diff
@@ -0,0 +1,41 @@
+#!/bin/bash
+
+# This script is used by the maven build (see pom.xml) to generate
+# translation source files based on the gettext .po files.
+
+# Fail fast to make sure we don't silently ignore unsuccessful msgfmt invocations (or other errors).
+set -e
+
+INPUT=po
+OUTPUT=target/generated-sources/msgfmt/gen/java
+
+# Create a temporary directory that we can freely munge with temporary build artifacts
+BUILD_DIR=$(mktemp -d -q msgfmt.XXXXXX)
+
+# Ensure our output directory exists
+mkdir -p $OUTPUT
+echo "Writing i18n catalog source files to: $OUTPUT"
+
+# Impl note:
+# msgfmt does not allow us to convert more than one .po file at a time due to the need to
+# specify the locale directly for the Java command. For this reason we do the initial
+# generation in the build directory and then copy the artifact the output directory.
+translation_files=`ls $INPUT/*.po`
```

Comment:
We can just use the directory/path directly for the for loop here:

```
for po_file in $INPUT/*.po
do
  ...
```

Reply:
done

Reply:
This still has the ls bit in there. The subshell and ls can be dropped by just specifying the path glob: `for po_file in $INPUT/*.po`

Note that if you quote the glob it no longer gets processed and the loop treats it as a single item.

Reply:
fixed